saleor/account/i18n.py

```diff
@@ -122,6 +123,7 @@ class Meta:
     def __init__(self, *args, **kwargs):
         autocomplete_type = kwargs.pop('autocomplete_type', None)
         super().__init__(*args, **kwargs)
+        self.fields['country'].choices = countries
```

Comment:
Why does this work?

Reply:
I think the countries order was taken as defined in the model, not being sorted accordingly to the selected language